tests/s25Main/UI/testImageButton.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2005 - 2022 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "controls/ctrlImageButton.h"
+#include "ogl/ITexture.h"
+#include "uiHelper/uiHelpers.hpp"
+#include <rttr/test/random.hpp>
+#include <turtle/mock.hpp>
+#include <boost/test/unit_test.hpp>
+
+using namespace rttr::test;
+
+namespace {
+MOCK_BASE_CLASS(TestTexture, ITexture)
+{
+public:
+    TestTexture(Extent size) : size_(size) {}
+    Position GetOrigin() const override { return Position::all(0); }
+    Extent GetSize() const override { return size_; }
+    MOCK_NON_CONST_METHOD(DrawFull, 2) // LCOV_EXCL_LINE
+    MOCK_NON_CONST_METHOD(Draw, 3)
+
+    Extent size_;
+};
+} // namespace
+
+// LCOV_EXCL_START
+namespace mock {
+static mock::stream& operator<<(mock::stream& s, const Rect& r)
+{
+    auto size = r.getSize();
+    return s << '(' << mock::format(r.left) << ',' << mock::format(r.top) << ',' << mock::format(size.x) << ','
+             << mock::format(size.y) << ')';
+}
+} // namespace mock
+// LCOV_EXCL_STOP
+
+BOOST_FIXTURE_TEST_SUITE(ImageButton, uiHelper::Fixture)
+
+BOOST_AUTO_TEST_CASE(DrawSmallerImageThanButton)
+{
+    TestTexture img(Extent(20, 10));
+    MOCK_EXPECT(img.Draw).once().with(Rect(20, 15, 20, 10), Rect(0, 0, 20, 10), 0xFFFFFFFFu);
+
+    Window wnd(nullptr, randomValue<unsigned>(), DrawPoint::all(0));
```

Comment:
I'd suggest to move the Window constructor to the top, so the values for the button, img and mock-expectation are closer together. I'd also suggest to place the `MOCK_EXPECT` right before the `draw()` just to make it more clear.